File: ex07/ex07_03/ex07_03.js
import fs from 'fs'
import promptSync from 'prompt-sync';
const prompt = promptSync();
const fileName = "../data/ex07_03.txt";
const upper = 100;
const lower = 60;
let data = [];

function main() {
  const dataSize = parseInt(prompt("Enter data size: "));
  if (isNaN(dataSize) || dataSize <= 0) {
    console.error("Invalid data size.");
    process.exit(1);
  }
  data.push(dataSize);
  for (let i = 0; i < dataSize; i++) {
    data.push(Math.floor(Math.random() * (upper - lower + 1)) + lower);
  }
  fs.writeFile(fileName, data.join('\n'), 'utf8', (err) => { 
      if (err) { 
        console.error('Error writing file:', err); 
        return; 
      }
  });
}
main();

File: ex07/ex07_03/ex07_03.py
import random
def main():
    file_name = "../data/ex07_03.txt"
    upper = 100
    lower = 60
    data = []
    data_size = int(input("Enter data size: "))
    if data_size <= 0:
        print("Invalid data size")
        return 1
    for i in range(data_size):
        data.append(random.randint(lower, upper))
    data.insert(0, data_size)
    lines = '\n'.join(map(str, data))
    fp = open(file_name, "w")
    fp.write(lines)
    fp.close()
    return 0    

if __name__ == "__main__":
    main()